userspace/libsinsp/sinsp_ppm_sc.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include <sinsp.h>
+
+std::unordered_set<uint32_t> sinsp::simple_ppm_sc_set()
+{
+	return enforce_sinsp_state_ppm_sc(std::unordered_set<uint32_t>{
+		PPM_SC_ACCEPT,
+		PPM_SC_ACCEPT4,
+		PPM_SC_BIND,
+		PPM_SC_BPF,
+		PPM_SC_CAPSET,
+		PPM_SC_CHDIR,
+		PPM_SC_CHMOD,
+		PPM_SC_CHROOT,
+		PPM_SC_CLONE,
+		PPM_SC_CLONE3,
+		PPM_SC_CLOSE,
+		PPM_SC_CONNECT,
+		PPM_SC_CREAT,
+		PPM_SC_DUP,
+		PPM_SC_DUP2,
+		PPM_SC_DUP3,
+		PPM_SC_EVENTFD,
+		PPM_SC_EVENTFD2,
+		PPM_SC_EXECVE,
+		PPM_SC_EXECVEAT,
+		PPM_SC_FCHDIR,
+		PPM_SC_FCHMOD,
+		PPM_SC_FCHMODAT,
+		PPM_SC_FCNTL,
+		PPM_SC_FCNTL64,
+		PPM_SC_FLOCK,
+		PPM_SC_FORK,
+		PPM_SC_GETSOCKOPT,
+		PPM_SC_INOTIFY_INIT,
+		PPM_SC_INOTIFY_INIT1,
+		PPM_SC_IOCTL,
+		PPM_SC_IO_URING_SETUP,
+		PPM_SC_KILL,
+		PPM_SC_LINK,
+		PPM_SC_LINKAT,
+		PPM_SC_LISTEN,
+		PPM_SC_MKDIR,
+		PPM_SC_MKDIRAT,
+		PPM_SC_MOUNT,
+		PPM_SC_OPEN,
+		PPM_SC_OPEN_BY_HANDLE_AT,
+		PPM_SC_OPENAT,
+		PPM_SC_OPENAT2,
+		PPM_SC_PIPE,
+		PPM_SC_PIPE2,
+		PPM_SC_PRLIMIT64,
+		PPM_SC_PTRACE,
+		PPM_SC_QUOTACTL,
+		PPM_SC_RECVFROM,
+		PPM_SC_RECVMSG,
+		PPM_SC_RENAME,
+		PPM_SC_RENAMEAT,
+		PPM_SC_RENAMEAT2,
+		PPM_SC_RMDIR,
+		PPM_SC_SECCOMP,
+		PPM_SC_SENDMSG,
+		PPM_SC_SENDTO,
+		PPM_SC_SETGID,
+		PPM_SC_SETGID32,
+		PPM_SC_SETNS,
+		PPM_SC_SETPGID,
+		PPM_SC_SETRESGID,
+		PPM_SC_SETRESGID32,
+		PPM_SC_SETRESUID,
+		PPM_SC_SETRESUID32,
+		PPM_SC_SETRLIMIT,
+		PPM_SC_SETSID,
+		PPM_SC_SETUID,
+		PPM_SC_SETUID32,
+		PPM_SC_SHUTDOWN,
+		PPM_SC_SIGNALFD,
+		PPM_SC_SIGNALFD4,
+		PPM_SC_SOCKET,
+		PPM_SC_SOCKETPAIR,
+		PPM_SC_SYMLINK,
+		PPM_SC_SYMLINKAT,
+		PPM_SC_TGKILL,
+		PPM_SC_TIMERFD_CREATE,
+		PPM_SC_TKILL,
+		PPM_SC_UMOUNT,
+		PPM_SC_UMOUNT2,
+		PPM_SC_UNLINK,
+		PPM_SC_UNLINKAT,
+		PPM_SC_UNSHARE,
+		PPM_SC_USERFAULTFD,
+		PPM_SC_VFORK,
+	});
+}
+
+std::unordered_set<uint32_t> sinsp::io_ppm_sc_set()
+{
+	std::unordered_set<uint32_t> ppm_sc_set;
+	for(int i = 0; i < PPM_SC_MAX; i++)
+	{
+		if(g_infotables.m_syscall_info_table[i].category == EC_IO_READ ||
```

Comment:
Also, @gnosek , `syscall_info_table` is not that useful nowadays: category can be fetched from `event_table`, flags are always `EF_NONE` and then there is name.
If we could find a way to fetch the correct name, we could at least build the table automatically when scap starts, to avoid having to remember to add data there.

It is useful when an GENERIC events is pushed onto the ring, because we push the `ppm_sc` as a first param, therefore having a `ppm_sc`-indexed table is helpful, but we can definitely build it programmatically IMHO.

Reply:
when do we end up with generic events? aren't they there for unsupported syscalls (for which we wouldn't have any details like category anyway)

Reply:
we use `GENERIC` also for some known syscalls like `epoll_ctl` and yes unfortunately we send both (scap id  and native id) to userspace

See here:
https://github.com/falcosecurity/libs/blob/master/driver/bpf/fillers.h#L2925-L2933